pkg/apis/v1alpha1/staticroute_types.go

```diff
@@ -0,0 +1,77 @@
+/* Copyright © 2022 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type StaticRouteStatusCondition string
+
+const (
+	StaticRouteReady    StaticRouteStatusCondition = "Ready"
+	StaticRouteNotReady StaticRouteStatusCondition = "NotReady"
+)
+
+// StaticRouteCondition defines condition of StaticRoute.
+type StaticRouteCondition struct {
+	// Type defines condition type.
+	Type StaticRouteStatusCondition `json:"type"`
+	// Status defines status of condition type, True or False.
+	Status corev1.ConditionStatus `json:"status"`
+	// Reason shows a brief reason of condition.
+	Reason string `json:"reason,omitempty"`
+	// Message shows a human readable message about condition.
+	Message string `json:"message,omitempty"`
+}
+
+// StaticRouteSpec defines static routes configuration on VPC
+type StaticRouteSpec struct {
+	// Specify next hop routes for network
+	StaticRoutes []StaticRouteItem `json:"routes"`
+}
+
+type StaticRouteItem struct {
+	// Unique route name to indicate the specific node in specific TKG
```

Comment:
We should follow NSX-T API. Do not talk about TKG, node, etc., which is too specific to the TKG use case.

Reply:
This is the static-route API request body. We can use the same fields instead of "target" and "destination".
{
      "network": "45.1.1.0/24",
      "next_hops": [
        {
          "ip_address": "41.1.1.1",
          "admin_distance": 1
        },
        {
          "ip_address": "42.1.1.1",
          "admin_distance": 2
        },
        {
          "ip_address": "43.1.2.3",
          "admin_distance": 3
        }
      ]
}

@heypnus , @jianjuns I see we only define 1 next hop for each network. Do we need be consistent with NSX API?

Reply:
Better to be consistent with NSX-T API.

Reply:
Now I updated so one staticroute has a nexthop list.